### PR TITLE
fix(2303): build cluster queue worker documentation

### PR DESCRIPTION
**Pull request by parthasl**
## Context

Cluster admins are not able to configure the build cluster queue worker due to missing documentation.

## Objective

Add build cluster queue worker documentation including installation and configuration of all involved components.

## References

https://github.com/screwdriver-cd/screwdriver/issues/2303

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — APPROVED** [review]

File: docs/cluster-management/configure-buildcluster-queue-worker.md [hunk 1]
```diff
@@ -0,0 +1,746 @@
+---
+layout: main
+title: Configuring Build Cluster Queue Worker
+category: Cluster Management
+menu: menu
+toc:
+    - title: Managing the Build Cluster Queue Worker 
+      url: "#manage-buildcluster-queue-worker"
```
Comment (tkyi) [inline docs/cluster-management/configure-buildcluster-queue-worker.md]:
```suggestion
      url: "#managing-the-build-cluster-queue-worker"
```

File: docs/cluster-management/configure-buildcluster-queue-worker.md [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+      active: true
+    - title: Overview
+      url: "#overview"
+    - title: Setup Build Cluster
+      url: "#setup-buildcluster"
```
Comment (tkyi) [inline docs/cluster-management/configure-buildcluster-queue-worker.md]:
```suggestion
      url: "#setup-build-cluster"
```

File: docs/cluster-management/configure-buildcluster-queue-worker.md [hunk 3]
```diff
@@ -0,0 +1,746 @@
...
+    - title: Install RabbitMQ Message Broker
+      url: "#install-rabbitmq"
+    - title: RabbitMQ Configuration
+      url: "#configure-rabbitmq"
+    - title: Configure Build Cluster Queue Worker
+      url: "#configure-buildcluster-queue-worker"
+    - title: Build Cluster Schema Definition
+      url: "#schema-definition"
```
Comment (tkyi) [inline docs/cluster-management/configure-buildcluster-queue-worker.md]:
```suggestion
      url: "#install-rabbitmq-message-broker"
    - title: RabbitMQ Configuration
      url: "#rabbitmq-configuration"
    - title: Configure Build Cluster Queue Worker
      url: "#configure-build-cluster-queue-worker"
    - title: Build Cluster Schema Definition
      url: "#build-cluster-schema-definition"
```

File: docs/cluster-management/configure-buildcluster-queue-worker.md [hunk 4]
```diff
@@ -0,0 +1,753 @@
...
+  "managedByScrewdriver": true,
+  "maintainer": "foo@bar.com",
+  "weightage": 100
+}
+```
+
+1. name should match the queue and routing key defined in rabbitmq configuration.
+1. set `managedByScrewdriver` to `true`, if build cluster is managed by internal Screwdriver team.
+1. set `isActive` to `true` or `false` to turn on/off a build cluster.
+1. set weightage 100 if you have single build cluster and for more than one cluster, distribute weightage accordingly.
```
Comment (s-yoshika) [inline docs/cluster-management/configure-buildcluster-queue-worker.md]: we had an [issue](https://github.com/screwdriver-cd/screwdriver/issues/2307) for multiple scms in multiple build clusters.
The [scmContext](https://github.com/screwdriver-cd/data-schema/blob/master/models/buildCluster.js#L30) of buildClusters is decided by the API Token which is belonging to the specific scm.
And if we want to build in the build cluster with another scm, we have to post `/v4/buildClusters` with same data by the another token which is issued by another scm account. it’s a little confusing so i'm glad if it has a caution for that.

Comment (parthasl) [reply docs/cluster-management/configure-buildcluster-queue-worker.md]: doc updated

**Review by s-yoshika — APPROVED** [review]

**Pull request merged**